src/bap_mc/bap_mc.ml

```diff
@@ -103,30 +110,41 @@ let print_bil lift bil_formats mem insn =
   List.iter bil_formats ~f:(fun fmt ->
       printf "%s@." (string_of_bil fmt (bil insn)))
 
-let make_print lift insn_fmt bil_fmt show_kinds mem insn =
-  print_insn insn_fmt insn;
+let make_print lift insn_fmt bil_fmt show_kinds oplen_bases mem insn =
+  print_insn_oplen oplen_bases mem;
+  print_insn insn_fmt insn; 
   print_bil lift bil_fmt mem insn;
   if show_kinds then print_kinds insn
 
-let step print state mem insn _ =
-  print mem insn;
+
+let check disasm_len current_total =
+  if disasm_len=0 then true
+  else current_total < disasm_len
+
+let step disasm_len hit_count print state mem insn _ =
+  if (check disasm_len !hit_count) then (
+    hit_count := !hit_count+1;
+    print mem insn;
+  ) else ();
```

Comment:
The compiler didn't admit it if just the else (); part was taken off. 

Thanks, I was searching for a way to halt the disassembler, but then thought this was a workaround, because in our use case we only pass 32 bytes at most. The thinking was "whatever is not printed is not decoded". But I agree with you, that is correct and the best way to go about it.

Thank you for the criticism.